lib/unleash/constraint.rb

```diff
@@ -106,8 +108,13 @@ def matches_constraint?(context)
       v = self.value.dup
       context_value = context.get_by_name(self.context_name)
 
-      v.map!(&:upcase) if self.case_insensitive
-      context_value = context_value.upcase if self.case_insensitive
+      # always return false, if we are comparing a non string with a string operator:
+      return false if !context_value.is_a?(String) && STRING_OPERATORS.include?(self.operator)
+
+      if self.case_insensitive
```

Comment:
The magic is now in the line above isn't it? This can change can get rolled back?

Reply:
Indeed the fix is in the line above. ( https://github.com/Unleash/unleash-client-ruby/pull/191/files#diff-aa0ebe0760222c549034e2dfab10a550f29c632fcc9919cc84cb0e7675a46409R112 )

These changes are mostly a lint/simplification. ( https://github.com/Unleash/unleash-client-ruby/pull/191/files#diff-aa0ebe0760222c549034e2dfab10a550f29c632fcc9919cc84cb0e7675a46409R114-R117 ). So, nice to have, but not fixing a problem per-se.